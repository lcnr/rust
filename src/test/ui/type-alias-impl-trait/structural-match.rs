#![feature(type_alias_impl_trait)]

type Foo = impl Send;

// This is not structural-match
struct A;

const fn value() -> Foo {
    A
}
const VALUE: Foo = value();

fn test() {
    match VALUE {
        VALUE => (),
        //~^ ERROR constant pattern depends on a generic parameter
        //~| ERROR constant pattern depends on a generic parameter
        _ => (),
    }
}

fn main() {}
