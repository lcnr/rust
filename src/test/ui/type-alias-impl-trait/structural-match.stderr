error: constant pattern depends on a generic parameter
  --> $DIR/structural-match.rs:15:9
   |
LL |         VALUE => (),
   |         ^^^^^

error: constant pattern depends on a generic parameter
  --> $DIR/structural-match.rs:15:9
   |
LL |         VALUE => (),
   |         ^^^^^

error: aborting due to 2 previous errors

